addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -342,6 +342,38 @@ def _import_retrieve_and_fill_partner(self, invoice, name, phone, mail, vat, cou
             if vat and self.env['res.partner']._run_vat_test(vat, country, invoice.partner_id.is_company):
                 invoice.partner_id.vat = vat
 
+    def _import_retrieve_and_fill_partner_bank_details(self, invoice, bank_details):
+        """ Retrieve the bank account, if no matching bank account is found, create it
+        """
+        if invoice.journal_id.type == 'purchase':
+            partner = invoice.partner_id
+        elif invoice.journal_id.type == 'sale':
+            partner = self.env.company.partner_id
+        else:
+            return
+
+        banks_to_create = []
+
+        for account_number in bank_details:
+            if account_number:
+                partner_bank = self.env['res.partner.bank'].search(
+                    [('company_id', 'in', [False, self.env.company.id]), ('acc_number', '=', account_number)],
+                    limit=1
+                )
+
+                if partner_bank:
+                    if partner_bank.partner_id.id == partner.id:
+                        invoice.partner_bank_id = partner_bank
+                        return
+                else:
+                    banks_to_create.append({
+                        'acc_number': account_number,
+                        'partner_id': partner.id,
+                    })
```

Comment:
Could this work?
```suggestion
                if partner_bank.partner_id == partner:
                        invoice.partner_bank_id = partner_bank
                        return
                else:
                    banks_to_create.append({
                        'acc_number': account_number,
                        'partner_id': partner.id,
                    })
```
or maybe it would be better to include the partner in the domain?

Reply:
No, let's imagine the case where the account number exists but not for the correct partner, and no other account are correct. In such situation, we would attempt to create a bank account with a bank number that already exists, this triggers an error because account numbers are meant to be unique.

Reply:
The same problem occurs if we add the partner_id in the search domain.. An bank account with the correct account number and wrong partner will not be returned by the search, if no other accounts are valid, we will then attempt to create a bank account with an account number that already exists.

Reply:
> No, let's imagine the case where the account number exists but not for the correct partner, and no other account exists and can be used directly. In such situation, we would attempt to create a bank account with a bank number that already exists, this triggers an error because account numbers are meant to be unique.

`elif partner_bank:`